draft-ietf-jsonpath-base.md

```diff
@@ -1443,17 +1443,28 @@ function-argument   = literal /
                       function-expr
 ~~~
 
-Any function expressions in a query must be well formed (by conforming to the above ABNF)
-and well typed,
+Any function expressions in a query must be well formed (by conforming to the above ABNF),
+supported, and well typed,
 otherwise the JSONPath implementation MUST raise an error
 (see {{synsem-overview}}).
+
+For a function expression to be supported, each function extension involved in the expression
+must be supported.
+Implementations MUST support the function extensions defined
+in this document, but are free to support or not support other registered function extensions.
+
+Implementations MUST NOT support function extensions that are not registered.
```

Comment:
This is not a very realistic requirement.
In particular during the time from developing a function extension to stabilizing it and making the specification widely available.

Reply:
My thinking was that during that period of time, the implementation would no longer conform to the spec. But it then becomes tricky to talk about that situation in the spec, as the back-and-forth between Greg and I has shown.

So perhaps this PR should be abandoned and I/we have to accept that there will be a free for all of custom function extensions, some of which may be registered and others of which will never get that far.